randomizer.py

```diff
@@ -1485,13 +1486,50 @@ def RandomizeDominion(setNames=None, options=None):
             counter += 1
 
         # Get final list of landscape cards
+        landscapeList = random.sample(waySet, len(waySet))[:1]
+        """Figure out card for Way of the Mouse. This uses similar rules to Young
+        Witch, so select a card from the Bane Cards. The card chosen for Way of the
+        Mouse should not be used when determining most additional card rules, but
+        we do need to check if it is a Liaison for ALly inclusion."""
+        includeMouse = Menagerie.cards("Way of the Mouse").intersection(landscapeList)
+        mouseSet = set()
+        if includeMouse:
+            eligibleMice = (kingdomSet & BaneCards) - resultSet
+            if not eligibleMice:
+                # All eligible Mouse cards are already part of the randomized set!
+                # (This is nearly impossible.) Get a Mouse from the randomized
+                # cards, add a new card to the set, and remove the mouse from the
+                # set.
+                mouseCard = random.sample(eligibleMice, 1)[0]
+                resultSet.update(random.sample(kingdomSet - resultSet, 1))
+                resultSet.remove(mouseCard)
+            else:
+                mouseCard = random.sample(eligibleMice, 1)[0]
+            mouseSet.add(mouseCard)
+        # Young Witch support
+        includeBane = resultSet & Cornucopia.cards("Young Witch")
+        if includeBane:
+            eligibleBanes = (kingdomSet & BaneCards) - resultSet - mouseSet
+            if not eligibleBanes:
+                # All eligible Bane cards are already part of the randomized set!
+                # Add a new card to the set and pull a Bane from the randomized
+                # cards.
+                resultSet.update(random.sample(kingdomSet - resultSet - mouseSet, 1))
+                baneCard = random.sample(resultSet & BaneCards - mouseSet, 1)[0]
+            else:
+                baneCard = random.sample(eligibleBanes, 1)[0]
+                resultSet.add(baneCard)
+        # Check for Liaisons (for a random Ally Card)
+        includeAlly = LiaisonCards & (resultSet | mouseSet)
+        if includeAlly:
+            ally = random.sample(AllyCards, 1)
+            landscapeList.extend(ally)
         if options and options.get("limit-landscapes"):
```

Comment:
There should be a blank line before this. Also, this would be a good place to put the `# Get final list of landscape cards` comment.